src/main/java/com/gromoks/filestatistic/service/statistic/FileStatistic.java

```diff
@@ -0,0 +1,124 @@
+                shortestWords = lineInfo.getShortestWords();
+            } else {
+                if (lineInfo.getLongestWords().get(0).length() > longestWords.get(0).length()) {
+                    longestWords = lineInfo.getLongestWords();
+                } else if (lineInfo.getLongestWords().get(0).length() == longestWords.get(0).length()) {
+                    longestWords.addAll(lineInfo.getLongestWords());
+                }
+
+                if (lineInfo.getShortestWords().get(0).length() < shortestWords.get(0).length()) {
+                    shortestWords = lineInfo.getShortestWords();
+                } else if (lineInfo.getShortestWords().get(0).length() == shortestWords.get(0).length()) {
+                    shortestWords.addAll(lineInfo.getShortestWords());
+                }
+            }
+        }
+        double average = new BigDecimal(averageSum / lineInfos.size()).setScale(2, RoundingMode.UP).doubleValue();
+
+        fileInfo.setFileName(fileName);
+        fileInfo.setLongestWords(longestWords);
+        fileInfo.setShortestWords(shortestWords);
+        fileInfo.setLength(length);
+        fileInfo.setAverageWordLength(average);
+
+        System.out.println(fileInfo);
+        return fileInfo;
+    }
+
+    static LineInfo getLineStatistic(String line, int rowNumber, String wordDelimiter) {
+        List<String> list = new ArrayList<>(Arrays.asList(line.split(wordDelimiter)));
+        LineInfo lineInfo = new LineInfo();
+
+        List<String> longestWords = new ArrayList<>();
+        List<String> shortestWords = new ArrayList<>();
+
+        int index = 0;
+        String longestWord = "";
+        String shortestWord = "";
+        int length = 0;
+        for (String row : list) {
+            index++;
+            if (index == 1) {
+                longestWord = row;
+                shortestWord = row;
+                longestWords.add(row);
+                shortestWords.add(row);
+            } else {
+                if (row.length() > longestWord.length()) {
+                    longestWord = row;
+                    longestWords.clear();
+                    longestWords.add(row);
+                } else if (row.length() == longestWord.length()) {
+                    longestWords.add(row);
+                } else if (row.length() < shortestWord.length()) {
+                    shortestWord = row;
+                    shortestWords.clear();
+                    shortestWords.add(row);
+                } else if (row.length() == shortestWord.length()) {
+                    shortestWords.add(row);
+                }
+            }
+            length = length + row.length();
+        }
+
+        lineInfo.setRowNumber(rowNumber);
+        lineInfo.setLongestWords(longestWords);
+        lineInfo.setShortestWords(shortestWords);
+        lineInfo.setRowLength(line.length());
+        lineInfo.setAverageWordLength(new BigDecimal((double) length / index).setScale(2, RoundingMode.UP).doubleValue());
```

Comment:
setScale(2, Round.UP) - Must be removed from avarage value calculation, because it has significant influance on calculation results